src/app/project/project-details-tab/project-details-tab.component.ts

```diff
@@ -13,14 +19,156 @@ import { NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
 export class ProjectDetailsTabComponent implements OnInit {
   public project;
   public commentPeriod = null;
+  public selectedCount = 0;
+  public loading = true;
+  public featuredDocuments: Document[] = null;
+  public documentTableData: TableObject;
+  public tableParams: TableParamsObject = new TableParamsObject();
+  public lists: any[] = [];
+  public documentTableColumns: any[] = [
+    {
+      name: 'Name',
+      value: 'displayName',
+      width: 'col-6'
+    },
+    {
+      name: 'Date',
+      value: 'datePosted',
+      width: 'col-2'
+    },
+    {
+      name: 'Type',
+      value: 'type',
+      width: 'col-2'
+    },
+    {
+      name: 'Milestone',
+      value: 'milestone',
+      width: 'col-2'
+    }
+  ];
+
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
 
   constructor(
+    private _changeDetectionRef: ChangeDetectorRef,
     private storageService: StorageService,
-    public configService: ConfigService
+    public configService: ConfigService,
+    private tableTemplateUtils: TableTemplateUtils,
+    private searchService: SearchService,
+    private route: ActivatedRoute,
+    private router: Router
   ) { }
 
   ngOnInit() {
     this.project = this.storageService.state.currentProject.data;
     this.commentPeriod = this.project.commentPeriodForBanner;
+
+    let params = null;
+    this.route.params
+      .switchMap((res: any) => {
+        params = { ...res };
+        return this.route.data;
+      })
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((res: any) => {
+
+      this.getFavoriteDocs();
+
```

Comment:
Looks like there might be an extra observable being passed around here. Could this just be
```this.route.params
      .subscribe((res: any) => {
        params = { ...res };
         this.getFavoriteDocs();
      })
```

Reply:
Yep, that's correct. Observable removed.